src/client/main.ts

```diff
@@ -416,16 +416,18 @@ function hasIntentListener(intent: string): boolean {
     return intentListeners.some((intentListener) => intentListener.intent === intent);
 }
 
-function deserializeChannelChangedEvent(eventTransport: Transport<ChannelChangedEvent>): ChannelChangedEvent {
-    const type = eventTransport.type;
-    const identity = eventTransport.identity;
-    const channel = eventTransport.channel ? getChannelObject(eventTransport.channel) : null;
-    const previousChannel = eventTransport.previousChannel ? getChannelObject(eventTransport.previousChannel) : null;
+if (typeof fin !== 'undefined') {
+    initialize();
 
-    return {type, identity, channel, previousChannel};
+    fin.InterApplicationBus.Channel.onChannelConnect((event: OpenFinChannelConnectionEvent) => {
```

Comment:
WE do almost the exact some thing in `connection`. Worth pulling out a shared function? Even if it's only somrthing that takes an event and returns true/false for whether it's about the service channel